File: backend/src/main/java/com/example/foodorderingapplication/services/Code/CodeService.java
package com.example.foodorderingapplication.services.Code;

import com.example.foodorderingapplication.dto.code.CodeDetails;
import com.example.foodorderingapplication.dto.code.CodeDto;

import java.util.List;

public interface CodeService {
    List<CodeDto> getRestaurantCodes(Long id);
    CodeDto createCode(Long id,CodeDetails codeDetails);
    CodeDto editCode(Long id,CodeDetails codeDetails);
    void deleteCode(Long id);
}

File: frontend/FoodOrderingApp/src/app/services/code.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {environment} from "../../environments/environment";
import {CodeDetails} from "../dto/code-details";
import {Observable} from "rxjs";
import {CodeDto} from "../dto/code-dto";

@Injectable({
  providedIn: 'root'
})

export class CodeService {

  codeURL: string;

  constructor(private http:HttpClient) {
    this.codeURL=`${environment.baseUrl}/code`
  }

  deleteCode(id:number):Observable<any>
  {
    return this.http.delete<any>(`${this.codeURL}/${id}`);
  }

  getDiscountCodes(id:number):Observable<CodeDto[]>
  {
    return this.http.get<CodeDto[]>(`${this.codeURL}/${id}`);
  }

  createCode(id:number,codeDetails:CodeDetails):Observable<CodeDto>
  {
    return this.http.post<CodeDto>(`${this.codeURL}/${id}`,codeDetails);
  }

  editCode(id:number,codeDetails:CodeDetails):Observable<CodeDto>
  {
    return this.http.put<CodeDto>(`${this.codeURL}/${id}`,codeDetails)
  }
}
